Fix IO.from_sql

dansbits opened the pull request:
The IO.from_sql method was using the dbi column_info to create the vectors of the data frame. This is resulting in an error:

```RuntimeError: can't add a new key into hash during iteration```

I've updated it to use the column_names method instead which does not result in updating the object you're iterating through.

v0dro commented on daru.gemspec:
```diff
@@ -51,6 +51,7 @@ EOF
 
   spec.add_runtime_dependency 'reportbuilder', '~> 1.4'
   spec.add_runtime_dependency 'spreadsheet', '~> 1.0.3'
+  spec.add_runtime_dependency 'dbi'
```
OK the rest seems fine but this runtime dependency on dbi will involve additional dependencies on daru itself, which I'd like to keep to a minimal.

What you can do is add it as a development dependency, and use something like the `Daru.has_gsl?` method before doing a `require` on dbi. You can throw an error asking the user to do `gem install dbi` if he wants to use the `from_sql` method.

See this code to see how you can add a `Daru.has_dbi?` method: https://github.com/v0dro/daru/blob/master/lib/daru.rb#L57

dansbits replied:
Moved this to a development dependency. I also removed the require 'dbi' from within the method. Because the user has to pass in a DBI::DatabaseHandle, it should be installed before calling from_sql.

v0dro commented on lib/daru/io/io.rb:
```diff
@@ -124,20 +124,24 @@ def dataframe_write_csv dataframe, path, opts={}
-        ds.update
-        ds
+
+        df
```
`df.update` will ensure that missing values are duly updated in the resulting dataframe. Removing it might lead to wrong results.

You can keep an option in the `from_sql` method, something like `:update_missing_values` and set the default to `true`. This can serve as a switch for updating the track of missing values that Daru::DataFrame keeps. This way people who dont care about the missing values being updated can disable the update from running.

An `if opts[:update_missing_values]` statement around the `df.update` will serve as the switch.

Also, don't forget to add documentation describing the new option. We follow YARD for docs. You can refer to other methods to see how it can be done.

dansbits replied:
I added a df.update. I didn't have any strong opinions about calling update, just wasn't sure if it was necessary.

v0dro commented on spec/io/io_spec.rb:
```diff
@@ -85,8 +85,26 @@
+        expect(accounts.nrows).to eq 2
+        expect(accounts.row[0][:id]).to eq 1
+        expect(accounts.row[0][:name]).to eq "Homer"
       end
```
Tests look great :)

v0dro commented:
Comments in code.

Also, could you rebase this feature into a single commit?

dansbits commented:
@v0dro I believe this is all taken care of. I squashed my commits into one and updated the yard docs for both DataFrame.from_sql and IO.from_sql. Just let me know if you have any other comments.

v0dro commented:
Thanks! Merging.
